src/Promitor.Core.Scraping/Configuration/Providers/MetricsDeclarationProvider.cs

```diff
@@ -32,9 +32,12 @@ public virtual MetricsDeclaration Get(bool applyDefaults = false)
                 foreach (var metric in config.Metrics)
                 {
                     // Apply AzureMetadata.ResourceGroupName to metrics with no other RG specified
-                    if (string.IsNullOrWhiteSpace(metric.ResourceGroupName))
+                    foreach (var resource in metric.Resources)
                     {
-                        metric.ResourceGroupName = config.AzureMetadata.ResourceGroupName;
+                        if (string.IsNullOrWhiteSpace(resource.ResourceGroupName))
+                        {
+                            resource.ResourceGroupName = config.AzureMetadata.ResourceGroupName;
```

Comment:
Since we already do a check in the scraper - WHat happens if we leave this out? Or can we assign the correct rg to scrape here already and get rid of it in the scraper?

Reply:
You're completely right - we don't actually need this because the logic is also in the Scraper. I've removed this and made the setter on `ResourceGroupName` private.